packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -47,67 +39,123 @@ describe("Dev component", () => {
       `);
     });
   });
+
+  describe("host", () => {
+    it("should resolve a host to its zone", async () => {
+      writeWranglerToml({
+        main: "index.js",
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev --host some-host.com");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone).toEqual({
+        host: "some-host.com",
+        id: "some-zone-id",
+      });
+    });
+
+    it("should read wrangler.toml's dev.host", async () => {
+      writeWranglerToml({
+        main: "index.js",
+        dev: {
+          host: "some-host.com",
+        },
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone.host).toEqual(
+        "some-host.com"
+      );
+    });
+
+    it("should read --route", async () => {
+      writeWranglerToml({
+        main: "index.js",
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev --route http://some-host.com/some/path/*");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone.host).toEqual(
+        "some-host.com"
+      );
+    });
+
+    it("should read wrangler.toml's routes", async () => {
+      writeWranglerToml({
+        main: "index.js",
+        routes: [
+          "http://some-host.com/some/path/*",
+          "http://some-other-host.com/path/*",
+        ],
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone.host).toEqual(
+        "some-host.com"
+      );
+    });
+
+    it("should read wrangler.toml's environment specific routes", async () => {
+      writeWranglerToml({
+        main: "index.js",
+        env: {
+          staging: {
+            routes: [
+              "http://some-host.com/some/path/*",
+              "http://some-other-host.com/path/*",
+            ],
+          },
+        },
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev --env staging");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone.host).toEqual(
+        "some-host.com"
+      );
+    });
+
+    it("given a long host, it should use the smallest domain that resolves to a zone", async () => {
```

Comment:
Are we actually testing for the smallest domain here?
It would seem that the only option is the last mock, which happens to be the smallest.

Reply:
OK, so looking at the code, I think this should say "longest" domain, right?